README.md
## PosterGen - News and Articles poster generator

Use for:
* Poster for your site or blog
* Poster for social network
* Poster for your promo materials &#9787;

## Example Images
![PosterGen](https://github.com/ange007/PosterGen/blob/master/icon.png) 

![PosterGen](https://github.com/ange007/PosterGen/blob/master/poster.png)

![PosterGen](https://github.com/ange007/PosterGen/blob/master/examples/1.png)

![PosterGen](https://github.com/ange007/PosterGen/blob/master/examples/2.png)

![PosterGen](https://github.com/ange007/PosterGen/blob/master/examples/3.png)

![PosterGen](https://github.com/ange007/PosterGen/blob/master/examples/4.png)

![PosterGen](https://github.com/ange007/PosterGen/blob/master/examples/5.png)

## Dependencies
* PHP >= 5.6
* GD

## Install
```
composer require ange007/poster-gen
```

## Example Code
```php
<?php
	use \PosterGen;
	
	// Generate poster
	$poster = ( new \PosterGen\PosterGen( [ ] ) )
		->setSize( 1280, 720 )
		->setBackgroundImage( __DIR__ . "/backgrounds/1.jpg" )
		->setHorizontalAlignment( 'center' )
		->setVerticalAlignment( 'center' )
		->setFontShadow( '#333333', -2, 2 )
		->setOverlayColor( '#FF0000' )
		->setBorder( 'black', 1 )
		// Title
		->setFont( __DIR__ . "/fonts/Roboto-Regular" )
		->setFontSize( 40 )
		->setFontColor( '#FFFFFF' )
		->addText( 'Microsoft buying GitHub' )
		->addText( '' )
		// Subtitle
		->setFont( __DIR__ . "/fonts/Blogger_Sans.otf" )
		->setFontSize( 20 )
		->setFontColor( '#00FFFF' )
		->addText( 'The deal is concluded' )
		// Watermark
		->setTextBackground( 'black', 50 )
		->setHorizontalAlignment( 'right' )
		->setVerticalAlignment( 'bottom' )
		->setFontSize( 14 )
		->setFontColor( '#FFFFFF' )
		->setFontShadow( '' )
		->setFontStroke( 'black' )
		->addText( 'http://news.com' );
		  
	// Poster output
	echo $poster->saveToBase64Image( );
```

## Plans
* Background position and adaptation
* Text rotate
* Image rotate

## License
It is released under the [MIT License](LICENSE).


CHANGELOG.md
### 0.6.5
* Improved calculation text block size.
* Improved line spacing.
* `setLinePadding` rename to `setLineSpacing`.
* Remove param `transparent` from `setBackgroundImage`.
* Small fixes and changes.

### 0.6.0
* Add text `style`: `underline`, `strike`.
* Add `addSeparator` function - for horizontal separation.
* Add `setBackgroundGradient` function - for background gradient.
* Add multi-line text support.

### 0.5.0
* First version.

src/Draw.php
<?php
namespace PosterGen;

trait Draw
{
	/**
	 * 
	 */
	protected function strokeText( $image, $values, $position, $color, $size )
	{
		if( $size <= 0 || $color === '' ) { return $this; }

		$x = $position[ 'x' ]; 
		$y = $position[ 'y' ];
		
		for( $c1 = $x - $size; $c1 <= $x + $size; $c1++ ) 
		{
			for( $c2 = $y - $size; $c2 <= $y + $size; $c2++ ) 
			{
				$this->drawInternal( $image, $values, [ 'x' => $c1, 'y' => $c2 ], $color );
			}
		}

		return $this;
	}

	/**
	 * @todo: https://stackoverflow.com/questions/14741622/underline-text-using-imagettftext
	 */
	protected function drawLine( $image, $position, $color )
	{
		$colorAllocate = $this->createColor( $image, $color );

		imageLine( $image, 
					$position[ 'x1' ], 
					$position[ 'y1' ], 
					$position[ 'x2' ], 
					$position[ 'y2' ], 
					$colorAllocate );

		return $this;
	}

	/**
	 * 
	 */
	protected function drawBorder( $image, $color, $size )
	{
		$colorAllocate = $this->createColor( $image, $color );

		for( $i = 0; $i < $size; $i++) 
		{
			imageRectangle( $image, 
							0 + $i, 
							0 + $i, 
							$this->size[ 'width' ] - ( 1 + $i ), 
							$this->size[ 'height' ] - ( 1 + $i ), 
							$colorAllocate );
		}

		return $this;
	}
	
	/**
	 * 
	 */
	protected function drawInternal( $image, $values, $position, $color, $transparent = 0 )
	{
		$colorAllocate = $this->createColor( $image, $color, $transparent );

		imageTTFText( $image, 
					$values[ 'font-size' ], 
					$values[ 'angle' ], 
					$position[ 'x' ], 
					$position[ 'y' ], 
					$colorAllocate,
					$values[ 'font' ], 
					$values[ 'text' ] );

		return $this;
	}

	/**
	 * 
	 */
	protected function drawBackgroundColor( $image, $position, $color, $transparent, $angle )
	{
		/* @todo: !! */
		if( $angle )
		{
			return $this;
		}

		$padding = 5;
		$colorAllocate = $this->createColor( $image, $color, $transparent );
		$transparencyColor = imageColorAllocateAlpha( $image , 0, 0, 0, 127 );

		// $this->imageFilledRotatedRectangle( $image, $position[ 'x' ], $position[ 'y' ], $position[ 'width' ], $position[ 'height' ], $angle, $colorAllocate );
		imageFilledRectangle( $image, 
							$position[ 'x' ] - $padding, 
							$position[ 'y' ] + $padding, 
							$position[ 'x' ] + $position[ 'width' ] + $padding,
							$position[ 'y' ] - $position[ 'height' ] - $padding, 
							$colorAllocate );
	}

	/**
	 * 
	 */
	protected function drawBackgroundImage( $image, $backgroundFile )
	{
		$backgroundImage = imageCreateFromString( file_get_contents( $backgroundFile ) );
		imageLayerEffect( $backgroundImage, IMG_EFFECT_NORMAL );
		imageSaveAlpha( $backgroundImage, true );

		//
		list( $backgroundImageWidth,  $backgroundImageHeight ) = getImageSize( $backgroundFile );
		$backgroundImageNewWidth = $backgroundImageWidth;
		$backgroundImageNewHeight = $backgroundImageHeight;	

		//
		if( $backgroundImageWidth < $this->size[ 'width' ]  )
		{
			$scalingFactor = $this->size[ 'width' ] / $backgroundImageWidth;
			$backgroundImageNewWidth = $this->size[ 'width' ];
			$backgroundImageNewHeight = round( $backgroundImageHeight * $scalingFactor );
		}
		// 
		else if( $backgroundImageHeight < $this->size[ 'height' ] )
		{
			$scalingFactor = $this->size[ 'height' ] / $backgroundImageWidth;
			$backgroundImageNewHeight = $this->size[ 'height' ];
			$backgroundImageNewWidth = round( $backgroundImageHeight * $scalingFactor );
		}

		//
		imageCopyResampled( $image, $backgroundImage, 
							0, 0, 
							0, 0,
							$backgroundImageNewWidth, $backgroundImageNewHeight, 
							$backgroundImageWidth, $backgroundImageHeight );

		//
		imageDestroy( $backgroundImage );
	}

	/**
	 * 
	 */
	protected function drawOverlay( $image, $color, $transparent )
	{
		$colorAllocate = $this->createColor( $image, $color, $transparent );
		$padding = 3;

		imageFilledRectangle( $image, 0, 0, $this->size[ 'width' ], $this->size[ 'height' ], $colorAllocate );

		return $this;	
	}

	/**
	 * 
	 */
	function generate( /*string*/ $format = 'png', $fileName = null )
	{
		//
		$image = imageCreateTrueColor( $this->size[ 'width' ], $this->size[ 'height' ] );
		imageLayerEffect( $image, IMG_EFFECT_NORMAL );
		imageSaveAlpha( $image, true );

		// Check for PHP < 7.2
		if( function_exists( 'imageAntialias' ) ) { imageAntialias( $image, True ); }

		// Colors
		$bgColor = $this->createColor( $image, $this->backgroundColor );

		//
		imageFillToBorder( $image, 0, 0, $bgColor, $bgColor );

		// Background Gradient
		if( is_array( $this->backgroundGradient ) && count( $this->backgroundGradient ) > 0 )
		{
			$this->imageFillGradient( $image, $this->backgroundGradient );
		}

		// Background Image
		if( !empty( $this->backgroundImage ) )
		{
			$this->drawBackgroundImage( $image, $this->backgroundImage );
		}

		// Overlay
		if( !empty( $this->overlayColor ) )
		{
			$this->drawOverlay( $image, $this->overlayColor, $this->overlayTransparent );
		}

		// Расчитываем высоту блоков
		$sectionsHeight = $this->calculateSectionsHeight( );

		//
		$verticalPadding = $this->verticalPadding + $this->borderSize;
		$horizontalPadding = $this->horizontalPadding + $this->borderSize;

		//
		$currentBottom = [ 
			'top' => $verticalPadding,
			'bottom' => $this->size[ 'height' ] - $sectionsHeight[ 'bottom' ] - $verticalPadding,
			'center' => ( ( $this->size[ 'height' ] - $sectionsHeight[ 'center' ] ) / 2.0 ) - $verticalPadding
		];

		// Debug
		if( $this->debug )
		{
			$this->paintDebugData( $image, $sectionsHeight, $currentBottom );
		}

		//
		$objectIndex = -1;

		// Start position
		foreach( $this->objectList as $value )
		{
			$objectIndex++;

			//
			$horizontalAlignment = array_get( $value, 'position.horizontal-alignment' );
			$verticalAlignment = array_get( $value, 'position.vertical-alignment' );

			// Horizontal position
			if( !empty( array_get( $value, 'position.x' ) ) ) { $left = array_get( $value, 'position.x' ); }
			else 
			{
				if( $horizontalAlignment === 'center' ) { $left = ( $this->size[ 'width' ] - array_get( $value, 'size.width' ) ) / 2.0; }
				else if( $horizontalAlignment === 'right' ) { $left = ( $this->size[ 'width' ] - array_get( $value, 'size.width' ) ) - $horizontalPadding; }
				else { $left = $horizontalPadding; }
			}

			// Image
			if( $value[ 'type' ] === 'image' )
			{
				// Vertical position
				if( !empty( array_get( $value, 'position.y' ) ) ) { $top = array_get( $value, 'position.y' ); }
				else
				{
					$top = $currentBottom[ $verticalAlignment ] - $this->lineSpacing;
					if( array_get( $value, 'position.vertical-alignment' ) === 'center' ) { $top += ( $this->size[ 'height' ] - array_get( $value, 'size.height' ) ) / 2.0; }
					else if( array_get( $value, 'position.vertical-alignment' ) === 'bottom' ) { $top += ( $this->size[ 'height' ] - array_get( $value, 'size.height' ) ) - $verticalPadding; }
					else { $top += $verticalPadding; }

					if( !$value[ 'inline' ] )
					{
						$currentBottom[ $verticalAlignment ] = $top + $this->lineSpacing;
					}
				}

				//
				$customImage = imageCreateFromString( file_get_contents( $value[ 'image' ] ) );
				imageLayerEffect( $customImage, IMG_EFFECT_NORMAL );
				imageSaveAlpha( $customImage, true );

				//
				$width = array_get( $value, 'size.width' );
				$height = array_get( $value, 'size.height' );

				//
				$customTop = ( ( $top + $height ) > $this->size[ 'height' ] ) ? $top + ( $this->size[ 'height' ] - ( $top + $height ) ) : $top;
				$customLeft = ( ( $left + $width ) > $this->size[ 'width' ] ) ? $left + ( $this->size[ 'width' ] - ( $left + $width ) ) : $left;

				// Copy image to canvas
				imageCopyResampled( $image,
									$customImage, 
									$customLeft, $customTop, 
									0, 0,
									$width, $height,
									array_get( $value, 'size.original-width' ), array_get( $value, 'size.original-height' ) );

				// 
				imageDestroy( $customImage );
			}
			// Text
			else if( $value[ 'type' ] === 'text' )
			{
				$charHeightDiffer = array_get( $value, 'coordinate.charHeightDiffer' );

				// Vertical position
				if( !empty( array_get( $value, 'position.y' ) ) ) { $top = array_get( $value, 'position.y' ); }
				else
				{
					$lineSpacing = $this->lineSpacing; 
					// $lineSpacing -= ( mb_strtolower( $value[ 'text' ] ) === $value[ 'text' ] ) ? $charHeightDiffer : 0;

					//
					$top = $currentBottom[ $verticalAlignment ];
					$top += ( $objectIndex === 0 ? 0 : $lineSpacing ) + array_get( $value, 'size.height' );

					//
					$currentBottom[ $verticalAlignment ] = $top;
				}

				//
				$stroke = $value[ 'stroke' ];
				$shadow = $value[ 'shadow' ];

				//
				if( empty( $value[ 'text' ] ) ) 
				{
					continue;
				}

				//
				if( $value[ 'text' ] === '-' )
				{
					$separatorWidth = $this->size[ 'width' ] / 2;
					// $separatorLeft = ( $this->size[ 'width' ] - $separatorWidth ) / 2;

					if( $horizontalAlignment === 'center' ) { $separatorLeft = ( $this->size[ 'width' ] - $separatorWidth ) / 2; }
					else if( $horizontalAlignment === 'right' ) { $separatorLeft = ( $this->size[ 'width' ] - $separatorWidth ); }
					else { $separatorLeft = $this->horizontalPadding + $this->borderSize; }

					$this->drawLine( $image, 
									[
										'x1' => $separatorLeft, 
										'x2' => $separatorLeft + $separatorWidth, 
										'y1' => $top - array_get( $value, 'coordinate.y' ) - ( array_get( $value, 'size.height' ) / 2 ),
										'y2' => $top - array_get( $value, 'coordinate.y' ) - ( array_get( $value, 'size.height' ) / 2 ),
									],
									$value[ 'color' ] );

					continue;
				}
			
				// Underline text
				if( is_array( $value[ 'style' ] ) && in_array( 'underline', $value[ 'style' ] ) )
				{
					$this->drawLine( $image, 
									[
										'x1' => $left, 
										'x2' => $left + array_get( $value, 'size.width' ), 
										'y1' => $top - array_get( $value, 'coordinate.y' ) + 2, 
										'y2' => $top - array_get( $value, 'coordinate.y' ) + 2
									],
									$value[ 'color' ] );
				}

				// Stroke text
				if( is_array( $stroke ) && count( $stroke ) > 0  )
				{
					$this->strokeText( $image, 
										$value, 
										[ 
											'x' => $left, 
											'y' => $top 
										], 
										$stroke[ 'color' ], 
										$stroke[ 'size' ] );
				}

				// Shadow
				if( is_array( $shadow ) && count( $shadow ) > 0 )
				{
					$this->drawInternal( $image, 
										$value,
										[ 
											'x' => $left + array_get( $shadow, 'offset.x', 0 ), 
											'y' => $top + array_get( $shadow, 'offset.y', 0 ) 
										], 
										$shadow[ 'color' ] );
				}

				// Text-line Background
				if( is_array( $value[ 'background' ] ) && !empty( array_get( $value, 'background.color' ) ) )
				{
					$this->drawBackgroundColor( $image, 
											[ 
												'x'	=> $left,
												'x1' => $left, 
												'x2' => $left + array_get( $value, 'size.width' ), 
												'y' => $top, 
												'y1' => $top - array_get( $value, 'coordinate.y' ), 
												'y2' => $top - array_get( $value, 'size.height' ),
												'height' => array_get( $value, 'size.height' ),
												'width' => array_get( $value, 'size.width' ) 
											], 
											array_get( $value, 'background.color' ), 
											array_get( $value, 'background.transparent' ),
											$value[ 'angle' ] );
				}

				// Draw text
				$this->drawInternal( $image,
									$value, 
									[ 
										'x' => $left, 
										'y' => $top
									], 
									$value[ 'color' ] );

				// Strike text
				if( is_array( $value[ 'style' ] ) && in_array( 'strike', $value[ 'style' ] ) )
				{
					$this->drawLine( $image,
									[
										'x1' => $left,
										'x2' => $left + array_get( $value, 'size.width' ),
										'y1' => $top - array_get( $value, 'coordinate.y' ) - ( array_get( $value, 'size.height' ) / 2 ),
										'y2' => $top - array_get( $value, 'coordinate.y' ) - ( array_get( $value, 'size.height' ) / 2 ),
									],
									$value[ 'color' ] );
				}
			}
		}

		// Border
		if( !empty( $this->borderColor ) && $this->borderSize > 0 )
		{
			$this->drawBorder( $image, $this->borderColor, $this->borderSize );
		}

		// 
		if( $format === 'jpeg' || $format === 'jpg' ) { imageJPEG( $image, $fileName ); }
		else if( $format === 'png' ) { imagePNG( $image, $fileName, 6, PNG_NO_FILTER  ); }
		else if( $format === 'gif' ) { imageGIF( $image, $fileName ); }

		//
		imageDestroy( $image );
	}
}

demo/ajax.php
<?php
	require_once '../src/Options.php';
	require_once '../src/Utils.php';
	require_once '../src/Draw.php';
	require_once '../src/PosterGen.php';

	function array_get( $array, $key, $default = null )
    {
        if( is_null( $key ) ) { return $array; }
		if( isset( $array[ $key ] ) ) { return $array[ $key ]; }
		
        foreach( explode( '.', $key ) as $segment )
        {
            if( !is_array( $array ) || !array_key_exists( $segment, $array ) )
            {
                return ( $default instanceof Closure ? $default( ) : $default );
			}
			
            $array = $array[ $segment ];
		}
		
        return $array;
    }

	/**
	 * 
	 */
	$request = $_REQUEST;
	$width = array_get( $request, 'width', 500 );
	$height = array_get( $request, 'height', 500 );

	/**
	 * 
	 */
	$poster = ( new \PosterGen\PosterGen( [ ] ) )
				->setSize( $width, $height )
				->setBackgroundColor( array_get( $request, 'background-color' ) )
				->setFont( __DIR__ . "/fonts/ComicRelief" );

	/**
	 * 
	 */
	$backgroundImage = array_get( $request, 'background-image' );
	if( !empty( $backgroundImage ) ) { $poster->setBackgroundImage( $backgroundImage ); }

	/**
	 * 
	 */
	for( $i = 0; $i < count( $request[ 'text' ] ); $i++ )
	{
		$text = array_get( $request[ 'text' ] , $i, '' );
		$size = array_get( $request[ 'size' ] , $i, 0 );
		$color = array_get( $request[ 'color' ] , $i, '' );

		$poster->addText( $text, '', $size, $color, [
			'position'	=> [
				'vertical-alignment' => array_get( $request[ 'vertical-alignment' ] , $i, '' ),
				'horizontal-alignment' => array_get( $request[ 'horizontal-alignment' ] , $i, '' ),
			]
		] );
	}

	/**
	 * 
	 */
	echo $poster->saveToBase64Image( );
?>


src/Options.php
<?php
namespace PosterGen;

trait Options
{
	protected $debug = false;
	protected $size = [ 'width' => 300, 'height' => 150 ];

	protected $font = '';
	protected $fontColor = '#000000';
	protected $fontSize = 14;
	protected $fontStyle = [ ];
	protected $lineSpacing = 8;

	protected $backgroundImage = '';
	protected $backgroundTransparent = 100;
	protected $backgroundColor = '#FFFFFF';

	protected $backgroundGradient = [ ]; // 4 colors ( 1 corner = 1 color )

	protected $overlayColor = '';
	protected $overlayTransparent = 100;

	protected $borderColor = '';
	protected $borderSize = 1;

	protected $strokeColor = '';
	protected $strokeSize = 1;

	protected $shadowColor = '';
	protected $shadowOffset = [ ];

	protected $textBackgroundColor = '';
	protected $textBackgroundTransparent = 0;

	protected $horizontalAlignment = 'center';
	protected $verticalAlignment = 'center';

	protected $verticalPadding = 10;
	protected $horizontalPadding = 10;

	protected $objectList = [ ];

	function __construct( array $options = [ ] )
	{

	}

	/**
	 * 
	 */

	function setSize( /*int*/ $width = 300, /*int*/ $sectionsHeight = 150 )
	{
		$this->size = [ 
			'width' => $width, 
			'height' => $sectionsHeight
		];

		return $this;
	}

	function getSize( /*bool*/ $realSize = False )
	{
		return [ 
			'width' => $this->size[ 'width' ] - ( $realSize ? 0 : $this->verticalPadding ), 
			'height' => $this->size[ 'height' ] - ( $realSize ? 0 : $this->horizontalPadding ), 
		];
	}

	function setFont( /*string*/ $value )
	{
		$fontFile = $value . ( empty( pathinfo( $value )[ 'extension' ] ) ? '.ttf' : '' );

		if( file_exists( $fontFile ) )
		{
			$this->font = $fontFile;
		}
		else
		{
			throw new \Exception( "PosterGen: No font available: {$fontFile}!" );
		}

		return $this;
	}

	function setFontColor( /*string*/ $value )
	{
		$this->fontColor = $value;

		return $this;
	}

	function setFontSize( /*int*/ $value )
	{
		$this->fontSize = $value;

		return $this;
	}

	function setFontStyle( array $value )
	{
		$this->fontStyle = $value;

		return $this;
	}

	function setFontShadow( $color, /*int*/ $x = 0, /*int*/ $y = 0 )
	{
		$this->shadowColor = $color;
		$this->shadowOffset = [ 'x' => $x, 'y' => $y ];

		return $this;
	}

	function setFontStroke( $color, /*int*/ $size = 1 )
	{
		$this->strokeColor = $color;
		$this->strokeSize = $size;
		
		return $this;
	}

	function setLineSpacing( /*int*/ $value )
	{
		$this->lineSpacing = $value;

		return $this;
	}

	function setVerticalPadding( /*int*/ $value )
	{
		$this->verticalPadding = $value;

		return $this;
	}

	function setHorizontalPadding( /*int*/ $value )
	{
		$this->horizontalPadding = $value;

		return $this;
	}

	function setBackgroundImage( /*string*/ $image )
	{
		if( file_exists( $image )
			|| ( filter_var( $image, FILTER_VALIDATE_URL ) && @get_headers( $image )[0] !== 'HTTP/1.0 404 Not Found' ) ) 
		{
			$this->backgroundImage = $image;
		}
		else
		{
			throw new \Exception( "PosterGen: No image available: {$image}!" );
		}

		return $this;
	}

	function setBackgroundColor( $color )
	{
		$this->backgroundColor = $color;

		return $this;
	}

	function setBackgroundGradient( array $colors )
	{
		$this->backgroundGradient = $colors;

		return $this;
	}

	function setOverlayColor( $color, /*int*/ $transparent = 40 )
	{
		$this->overlayColor = $color;
		$this->overlayTransparent = $transparent;

		return $this;
	}

	function setVerticalAlignment( /*string*/ $value )
	{
		$this->verticalAlignment = $value;

		return $this;
	}

	function setHorizontalAlignment( /*string*/ $value )
	{
		$this->horizontalAlignment = $value;

		return $this;
	}

	function setBorder( $color, /*int*/ $size )
	{
		$this->borderColor = $color;
		$this->borderSize = $size;

		return $this;	
	}

	function setTextBackground( $color, /*int*/ $transparent = 0 )
	{
		$this->textBackgroundColor = $color;
		$this->textBackgroundTransparent = $transparent;

		return $this;
	}

	function debug( /*bool*/ $value )
	{
		$this->debug = $value;

		return $this;
	}
}

demo/demo.php
<?php
	require_once '../src/Helpers.php';
	require_once '../src/Options.php';
	require_once '../src/Utils.php';
	require_once '../src/Draw.php';
	require_once '../src/PosterGen.php';
?>

<!DOCTYPE html>
<html lang="en">
	<head>
		<style>
			body {
				text-align: center;
				background: #74E868;
			}

			.code {
				max-height: 100px;
				overflow: scroll;
				display: inline-block;
			}

			img { 
				padding: 50px;
				border: 1px solid #14D100;
				margin: 10px;
				background: #CDF76F;
			}
		</style>
	</head>
	<body>
	<?php
			/**
			 * 
			 */
			$poster0 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 80, 80 )
						->setBackgroundColor( '#00BD39' )
						->setBackgroundImage( __DIR__ . "/backgrounds/1.jpg" )
						->setHorizontalAlignment( 'center' )
						->setVerticalAlignment( 'center' )
						->setBorder( 'black', 1 )
						->setOverlayColor( '#025167', 80 )
						->setFont( __DIR__ . "/fonts/Blogger_Sans.otf" )
						//
						->setFontSize( 14 )
						->setFontColor( '#FF5A40' )
						->setFontStroke( '#025167' )
						->addText( 'PosterGen' )
						//
						->setFontSize( 12 )
						->setFontColor( '#FFC640' )
						->addText( 'Poster' )
						->addText( 'generator' );

			echo $poster0->saveToBase64Image( );
		?>
		
		<br />

		<?php
			/**
			 * 
			 */
			$poster1 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 590, 300 )
						->setBackgroundColor( '#00BD39' )
						->setBackgroundImage( __DIR__ . "/backgrounds/1.jpg" )
						->setVerticalAlignment( 'top' )
						->setHorizontalAlignment( 'center' )
						->setVerticalPadding( 30 )
						->setHorizontalPadding( 30 )
						->setBorder( 'black', 1 )
						->setOverlayColor( '#025167', 80 )
						->setFont( __DIR__ . "/fonts/Blogger_Sans.otf" )
						//
						->setFontSize( 40 )
						->setFontColor( '#FF5A40' )
						->setFontStroke( '#025167' )
						->addText( 'PosterGen' )
						//
						->setFontSize( 23 )
						->setFontColor( '#FFC640' )
						->addText( 'Poster generator for all' )
						//
						->addSeparator( '', 'center' )
						//
						->setHorizontalAlignment( 'left' )
						->setVerticalAlignment( 'bottom' )
						->setFontSize( 19 )
						->setFontColor( '#39AECF' )
						->addText( '&#8226; Poster for your site or blog' )	
						->addText( '&#8226; Poster for social network' )
						->addText( '&#8226; Poster for your promo materials &#9787;' )
						//
						->setHorizontalAlignment( 'right' )
						->setTextBackground( '#007B25' )
						->setVerticalAlignment( 'bottom' )
						->setFontSize( 14 )
						->setFontColor( '#FFFFFF' )
						->addText( 'best for you!', '', 0, '', [ ], [ ] );

			echo $poster1->saveToBase64Image( );
		?>
		
		<br />

		<?php
			/**
			 * 
			 */
			$poster2 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 640, 640 )
						->setBackgroundImage( __DIR__ . "/backgrounds/4.jpg" )
						->setHorizontalAlignment( 'center' )
						->setVerticalAlignment( 'center' )
						->setOverlayColor( '#000000' )
						->setBorder( 'black', 1 )
						//
						->setFont( __DIR__ . "/fonts/Ge_Body" )
						->setFontSize( 40 )
						->setFontColor( '#FFFFFF' )
						->addText( 'Overwatch: New Hero Brigitte' )
						->addText( '' )
						//
						->setFontSize( 20 )
						->setFontColor( '#FF5A40' )
						->addText( 'Release Date' )
						//
						->setTextBackground( 'white', 0 )
						->setHorizontalAlignment( 'left' )
						->setVerticalAlignment( 'top' )
						->setFontSize( 32 )
						->setFontColor( 'black' )
						->setFontShadow( '' )
						->setFontStroke( 'black' )
						->addText( 'OVERWATCH' )
						//
						->setTextBackground( '' )
						->setHorizontalAlignment( 'left' )
						->setVerticalAlignment( 'bottom' )
						->setFontSize( 14 )
						->setFontColor( '#FFFFFF' )
						->setFontShadow( '' )
						->setFontStroke( 'black' )
						->addText( 'http://news.com' );

			echo $poster2->saveToBase64Image( );
		?>
			
		<br />

		<?php
			/**
			 * 
			 */
			$poster3 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 860, 300 )
						->setBackgroundImage( __DIR__ . "/backgrounds/3.jpg" )
						->setHorizontalAlignment( 'center' )
						->setVerticalAlignment( 'center' )
						->setOverlayColor( '#000000' )
						->setBorder( 'black', 1 )
						//
						->setFont( __DIR__ . "/fonts/CaviarDreams_Bold" )
						->setFontSize( 40 )
						->setFontColor( '#FFFFFF' )
						->addText( 'Windows 10' )
						//
						->setFontSize( 20 )
						->setFontColor( '#559ce4' )
						->addText( 'Review April 2018 Update' )
						//
						->setTextBackground( 'white', 0 )
						->setVerticalAlignment( 'top' )
						->setFontSize( 24 )
						->setFontColor( 'black' )
						->setFontShadow( '' )
						->setFontStroke( 'black' )
						->addText( 'News > Tech' )
						//
						->setTextBackground( 'black', 0 )
						->setHorizontalAlignment( 'right' )
						->setVerticalAlignment( 'bottom' )
						->setFontSize( 14 )
						->setFontColor( '#FFFFFF' )
						->setFontShadow( '' )
						->setFontStroke( 'black' )
						->addText( 'http://news.com' );

			echo $poster3->saveToBase64Image( );
		?>
		
		<br />
		
		<?php
			/**
			 * 
			 */
			$poster4 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 350, 200 )
						->setBackgroundColor( '#FFFFFF' )
						->setFontColor( '#9A0000' )
						->setFontSize( 26 )
						->setFontStroke( '#FFFFFF', 2 )
						->setBorder( '#9A0000', 2 )
						->setFont( __DIR__ . "/fonts/ComicRelief" )
						//
						->addImage( __DIR__ . "/images/clipboard.png", [ 'inline' => true ] )
						//
						->setVerticalAlignment( 'top' )
						->setHorizontalAlignment( 'left' )
						->addText( 'Documentation' )
						//
						->setVerticalAlignment( 'center' )
						->setHorizontalAlignment( 'center' )
						->addText( 'Instruction', '', 34 )
						//
						->setVerticalAlignment( 'bottom' )
						->setHorizontalAlignment( 'right' )
						//
						->addText( 'F.A.Q', '', 36 );

			echo $poster4->saveToBase64Image( );
		?>

		<?php
			/**
			 * 
			 */
			$poster5 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 350, 200 )
						->setBackgroundColor( '#FF5A40' )
						->setFontColor( '#FFC640' )
						->setFontSize( 24 )
						->setFontStroke( '#007B25' )
						->setFont( __DIR__ . "/fonts/ComicRelief" )
						->setVerticalAlignment( 'top' )
						//
						->addImage( __DIR__ . "/images/wink.png" )
						//
						->setVerticalAlignment( 'bottom' )
						->addText( 'Smiles improve health' );

			echo $poster5->saveToBase64Image( );
		?>

		<?php
			/**
			 * 
			 */
			$poster6 = ( new \PosterGen\PosterGen( [ ] ) )
						->setSize( 350, 200 )
						->setBackgroundGradient( [ '#FF5A40', '#007B25', '#FFC640', '#007B25' ] )
						->setFontColor( '#FFFFFF' )
						->setFontSize( 24 )
						->setFontStroke( '#007B25' )
						->setFont( __DIR__ . "/fonts/ComicRelief" )
						->addText( "Party on\r\nLasVegas" );

			echo $poster6->saveToBase64Image( );
		?>
	</body>
</html>
